vital/types/detected_object.h

```diff
@@ -80,6 +81,15 @@ class VITAL_EXPORT detected_object
   typedef descriptor_dynamic< double > descriptor_t;
   typedef std::shared_ptr< descriptor_t const > descriptor_scptr;
 
+  /**
+   * @brief Create default detected object.
+   *
+   * @param confidence Detectors confidence in this detection.
+   * @param classifications Optional object classification.
+   */
+  detected_object( double confidence = 1.0,
+                   detected_object_type_sptr classifications = detected_object_type_sptr());
```

Comment:
BTW, `= {}` or `= nullptr`? (Also other new ctor, and maybe change the existing one(s) while you're at it?)